### PR TITLE
(GH-2240) Document task parameter constraints

### PR DESCRIPTION
**Pull request by beechtom**
This adds documentation for constraints around task parameter names and
types.

!no-release-note

### CODE REVIEW

File: documentation/writing_tasks.md [hunk 1]
```diff
@@ -185,6 +185,18 @@ If your task must receive parameters only in a certain way, such as only
 `input_method`. For more information on `input_method`, see [Task metadata
 reference](#task-metadata-fields).
 
+### Task parameter constraints
+
+There are a couple important constraints to keep in mind when defining your
+task's parameters:
+
+- Task parameter names must begin with a lowercase letter, and may only include
```
Comment (hestonhoffman) [inline documentation/writing_tasks.md:193]:
```suggestion
- Task parameter names must begin with a lowercase letter, and can only include
```
😁

**Review by hestonhoffman — APPROVED** [review]
One nit, but looks good!

File: documentation/writing_tasks.md (hunk 1 above)
Comment (beechtom) [reply documentation/writing_tasks.md:193]: Noooo! 😂 I've been trying so hard not to use "may".

Comment (hestonhoffman) [reply documentation/writing_tasks.md]: You're doing great! This is kind of a tricky one cos you can't use might here. It's actually part of the reason we don't like"may" - the ambiguity for non-native speakers.

Comment (beechtom) [reply documentation/writing_tasks.md]: Makes total sense 👍  Thanks Heston!

**Pull request merged**